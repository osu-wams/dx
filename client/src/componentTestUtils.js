import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createHistory, createMemorySource, LocationProvider } from '@reach/router';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

function render(ui, ...rest) {
  return rtlRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, ...rest);
}

// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...rtlRender(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us to reference it in our tests
    history
  };
}

export * from 'react-testing-library';
// override the built-in render with our own
export { render, renderWithRouter };
