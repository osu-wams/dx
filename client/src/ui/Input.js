import styled from 'styled-components';
import PropTypes from 'prop-types';

// Outputs the padding size in pixel for the component
function inputSize(size, theme) {
  if (size === 'small') {
    return `
      padding: ${theme.spacing.unit / 4}px ${theme.spacing.unit / 2}px;
      font-size: ${theme.fontSize.small};
    `;
  }
  if (size === 'large') {
    return `
      padding: ${theme.spacing.unit * 2}px ${theme.spacing.unit * 4}px;
      font-size: ${theme.fontSize.large};
    `;
  }
  return `
      padding: ${theme.spacing.unit}px ${theme.spacing.unit * 2}px;
      font-size: ${theme.fontSize.normal};
    `;
}

const Input = styled.input`
  color: ${props => props.theme.colors.charcoal};
  text-decoration: none;
  border-radius: 0.4rem;
  border: 1px solid ${props => props.theme.colors.dusk};
  ${({ theme, size }) => inputSize(size, theme)};
`;

Input.propTypes = {
  size: PropTypes.oneOf(['small', 'normal', 'large'])
};

Input.defaultProps = {
  size: 'normal'
};

export default Input;
