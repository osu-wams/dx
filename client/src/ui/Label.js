import styled from 'styled-components';

const Label = styled.label`
  color: ${props => props.theme.colors.charcoal};
`;

export default Label;
