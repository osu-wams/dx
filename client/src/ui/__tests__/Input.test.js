import React from 'react';
import { render } from '../../componentTestUtils';
import Label from '../Label';
import Input from '../Input';

const SmallInput = () => (
  <div>
    <Label htmlFor="small">
      Small
      <Input type="text" id="small" size="small" />
    </Label>
  </div>
);

const NormalInput = () => (
  <div>
    <Label htmlFor="normal">
      Normal
      <Input type="text" id="normal" />
    </Label>
  </div>
);

const LargeInput = () => (
  <div>
    <Label htmlFor="large">
      Large
      <Input type="text" id="large" size="large" />
    </Label>
  </div>
);

test('SmallInput gets rendered with small font size CSS', () => {
  const { container } = render(<SmallInput />);
  expect(container.firstChild).toMatchSnapshot();
});

test('NormalInput gets rendered with default font size CSS', () => {
  const { container } = render(<NormalInput />);
  expect(container.firstChild).toMatchSnapshot();
});

test('LargeInput gets rendered with large font size CSS', () => {
  const { container } = render(<LargeInput />);
  expect(container.firstChild).toMatchSnapshot();
});
