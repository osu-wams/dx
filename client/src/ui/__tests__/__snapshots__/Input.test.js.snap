// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeInput gets rendered with large font size CSS 1`] = `
.c0 {
  color: #252525;
}

.c1 {
  color: #252525;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0.4rem;
  border: 1px solid #444;
  padding: 16px 32px;
  font-size: 1.8rem;
}

<div>
  <label
    class="c0"
    for="large"
  >
    Large
    <input
      class="c1"
      id="large"
      type="text"
    />
  </label>
</div>
`;

exports[`NormalInput gets rendered with default font size CSS 1`] = `
.c0 {
  color: #252525;
}

.c1 {
  color: #252525;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0.4rem;
  border: 1px solid #444;
  padding: 8px 16px;
  font-size: 1.6rem;
}

<div>
  <label
    class="c0"
    for="normal"
  >
    Normal
    <input
      class="c1"
      id="normal"
      type="text"
    />
  </label>
</div>
`;

exports[`SmallInput gets rendered with small font size CSS 1`] = `
.c0 {
  color: #252525;
}

.c1 {
  color: #252525;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0.4rem;
  border: 1px solid #444;
  padding: 2px 4px;
  font-size: 1.4rem;
}

<div>
  <label
    class="c0"
    for="small"
  >
    Small
    <input
      class="c1"
      id="small"
      type="text"
    />
  </label>
</div>
`;
